lib/PuppeteerSharp/Page.cs

```diff
@@ -1283,8 +1283,7 @@ public Task<ElementHandle> WaitForXPathAsync(string xpath, WaitForSelectorOption
             var watcher = new NavigatorWatcher(_frameManager, mainFrame, timeout, options);
             var responses = new Dictionary<string, Response>();
 
-            EventHandler<ResponseCreatedEventArgs> createResponseEventListener = (object sender, ResponseCreatedEventArgs e) =>
-                responses.Add(e.Response.Url, e.Response);
+            EventHandler<ResponseCreatedEventArgs> createResponseEventListener = (object sender, ResponseCreatedEventArgs e) => responses[e.Response.Url] = e.Response;
```

Comment:
Don't hate me 😂  But this could be shorter like this
```
EventHandler<ResponseCreatedEventArgs> createResponseEventListener = (sender, e) => responses[e.Response.Url] = e.Response;
```

Reply:
oh, I did not notice that because that is what the original code is.. I will make another change now, thanks~